src/System.Security.Cryptography.X509Certificates/src/Internal/Cryptography/Pal.Unix/ChainPal.cs

```diff
@@ -25,7 +25,26 @@ public static bool ReleaseSafeX509ChainHandle(IntPtr handle)
             DateTime verificationTime,
             TimeSpan timeout)
         {
-            return new OpenSslX509ChainProcessor();
+            CheckRevocationMode(revocationMode);
+
+            X509Certificate2 leaf = new X509Certificate2(cert.Handle);
+
+            X509Certificate2Collection candidates = OpenSslX509ChainProcessor.FindCandidates(leaf, extraStore);
+
+            return OpenSslX509ChainProcessor.BuildChain(
+                leaf,
+                candidates,
+                applicationPolicy,
+                certificatePolicy,
+                verificationTime);
+        }
+
+        private static void CheckRevocationMode(X509RevocationMode revocationMode)
+        {
+            if (revocationMode != X509RevocationMode.NoCheck)
+            {
+                throw new NotImplementedException(SR.WorkInProgress);
```

Comment:
TODO. It's dependent on a stable networking stack (since I can't make use of the OpenSSL functions on OSX's 0.9.8). So not beta6, but I'd feel really bad if this wasn't done by X-Plat RTM.